articles/quickstart/backend/java-spring-security5/interactive.md

```diff
@@ -0,0 +1,150 @@
+
+### Protect API Endpoints
+
+<%= include('../_includes/_api_endpoints') %>
+
+The example below shows how to secure API methods using the `HttpSecurity` object provided in the `configure()` method of the `SecurityConfig` class. Route matchers are used to restrict access based on the level of authorization required:
+
+::: note
+By default, Spring Security will create a `GrantedAuthority` for each scope in the `scope` claim of the JWT. This is what enables using the `hasAuthority("SCOPE_read:messages")` method to restrict access to a valid JWT that contains the `read:messages` scope.
+
+If your use case requires different claims to make authorization decisions, see the [Spring Security Reference Documentation](https://docs.spring.io/spring-security/site/docs/current/reference/htmlsingle/#oauth2resourceserver-authorization-extraction) to learn how to customize the extracted authorities.
+:::
+
+### Configure JWT Validator
+
+Update the `SecurityConfig` class to configure a `JwtDecoder` bean that uses the `AudienceValidator`. The `auth0.audience` value from `application.yml` will be used to validate the `aud` claim. The `issuer` is verified using the `issuer-uri` value from the `application.yml`
+
+## Create the Domain Object {{{ data-action=code data-code="Message.java#1:11" }}}
+
+Create a new class named `Message`. This will act as a simple domain object to return during the API calls.
+
+## Create the API controller {{{ data-action=code data-code="APIController.java" }}}
+
+Create a new class named `APIController` to handle requests to the endpoints. The `APIController` will have three routes as defined in the [Protect API Endpoints](/quickstart/backend/java-spring-security5/interactive/#configure-the-resource-server) section. For the simplicity of this sample we will allow all origins through `@CrossOrigin` annotation. Real applications should configure `CORS` for their use case.
+
+## Run the Application {{{ data-action=code data-code="APIController.java" }}}
+
+To build and run the sample project, execute the `bootRun` Gradle task.
+
+Linux or macOS:
+
+```bash
+./gradlew bootRun
+```
+
+Windows:
+
+```bash
+gradlew.bat bootRun
+```
+
+If you are configuring your own application using Maven and the [Spring Boot Maven Plugin](https://docs.spring.io/spring-boot/docs/current/reference/html/build-tool-plugins-maven-plugin.html), you can execute the `spring-boot:run` goal.
+
+Linux or macOS:
+
+```bash
+mvn spring-boot:run
+```
+
+Windows:
+
+```bash
+mvn.cmd spring-boot:run
+```
+
+The sample application will be available at `http://localhost:3010/`. Read about how to test and use your API in the [Using Your API](/quickstart/backend/java-spring-security5/02-using) article. Incase of unexpected responses from your API, use the [Troubleshooting](/quickstart/backend/java-spring-security5/03-troubleshooting) section to check your configuration.
```

Comment:
Can we add a checkpoint here ? For example as done here https://github.com/auth0/docs/pull/10085